org_fedora_oscap/rule_handling.py

```diff
@@ -938,48 +951,51 @@ def eval_rules(self, ksdata, storage, report_only=False):
 
         # services, that should be added
         services_to_add = (svc for svc in self._add_svcs
-                           if svc not in ksdata.firewall.services)
+                           if svc not in firewall_proxy.EnabledServices)
 
         # ports, that should be added
         ports_to_add = (ports for ports in self._add_ports
-                        if ports not in ksdata.firewall.ports)
+                        if ports not in firewall_proxy.EnabledPorts)
 
         # trusts, that should be added
         trusts_to_add = (trust for trust in self._add_trusts
-                         if trust not in ksdata.firewall.trusts)
+                         if trust not in firewall_proxy.EnabledTrusts)
 
         for svc in services_to_add:
             # add the service unless already added
             if not report_only:
                 self._added_svcs.add(svc)
-                ksdata.firewall.services.append(svc)
 
             msg = _("service '%s' has been added to the list of services to be "
                     "added to the firewall" % svc)
             messages.append(RuleMessage(self.__class__,
                                         common.MESSAGE_TYPE_INFO, msg))
+        if not report_only:
+            firewall_proxy.SetEnabledServices(self._add_svcs)
```

Comment:
I couldn't exactly find in documentation, but from [source code](https://github.com/rhinstaller/anaconda/tree/master/pyanaconda/modules/network/firewall), it looks like the new interface doesn't append, it just sets the new list of services.

So this should probably contain both the newly enabled and already enabled services.

Reply:
Yes the documentation is outdated for the current code, sorry for that. And again yes, as you said this should set a list not append.